zero/src/trie_diff/mod.rs

```diff
@@ -0,0 +1,111 @@
+use evm_arithmetization::generation::mpt::{AccountRlp, LegacyReceiptRlp};
+use evm_arithmetization::generation::DebugOutputTries;
+use mpt_trie::debug_tools::diff::create_diff_between_tries;
+use mpt_trie::utils::TrieNodeType;
+use tracing::info;
+
+use crate::prover::BlockProverInput;
+
+pub fn compare_tries(
+    block_prover_input: &BlockProverInput,
+    batch_index: usize,
+    left: &DebugOutputTries,
+    right: &DebugOutputTries,
+) -> anyhow::Result<()> {
+    let block_number = block_prover_input
+        .other_data
+        .b_data
+        .b_meta
+        .block_number
+        .low_u64();
+    let state_trie_diff = create_diff_between_tries(&left.state_trie, &right.state_trie);
+    if let Some(ref state_trie_diff_point) = state_trie_diff.latest_diff_res {
+        if state_trie_diff_point.a_info.node_type == TrieNodeType::Leaf {
+            if let Some(ref td_account_value) = state_trie_diff_point.a_info.value {
+                let td_account_data = rlp::decode::<AccountRlp>(td_account_value)?;
+                info!("Trace decoder state trie block {block_number} batch {batch_index} account address hash: {} account data: {:#?}",
```

Comment:
There's a lot of duplication in here that could probably be reduced to make it a little cleaner (maybe) but I can't say I care too much

Reply:
Indeed, I took the easier copy/paste path as this is just a debugging tool. Let me think if I can make it better.